File: src/generator/vue-admin/crawler/crawler/src/main/java/com/jasmine/crawler/pojo/entity/BindRecord.java
/*----------------------------------------------------------------------------
 * Jasmine code generator, a tool to build web crud application,with spring-
 * boot, mybatis, mysql,swagger,spring-security.
 * Generated at 2020-7-31 9:09:23 PM 
 * All rights reserved by fuanlei(email:767550758@qq.com) since 2019
 *---------------------------------------------------------------------------*/
package com.jasmine.crawler.pojo.entity;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@ApiModel("")
@Getter
@Setter
public class BindRecord  {

    @ApiModelProperty("")
    private Integer id;

    @ApiModelProperty("")
    private Integer crawlTaskId;

    @ApiModelProperty("")
    private Date createTime;

    @ApiModelProperty("")
    private String msg;

    @ApiModelProperty("")
    private Integer bindStatus;

}

File: src/generator/vue-admin/crawler/config/bindRecord.js
/*----------------------------------------------------------------------------
 * Jasmine code generator, a tool to build web crud application,with spring-
 * boot, mybatis, mysql,swagger,spring-security.
 * Generated at 2020-7-31 8:47:21 PM 
 * All rights reserved by fuanlei(email:767550758@qq.com) since 2019
 *---------------------------------------------------------------------------*/
const { all } = require("./../db/all")
const { ConfigItemBuilder } = require("C:/Users/Administrator/Desktop/Projects/code-generator/src/generator/spring-boot/builders/ConfigItemBuilder")


exports.bindRecordConfig = {
        table: all.bindRecord,
        name: "BindRecord",
                items: [
// getPage
                        // createTime : expression --- timeRange
                        // msg : excluded 
                        // bindStatus : validates --- @Enum("bindStatus")
                        new ConfigItemBuilder()
                                .type("select")
                                .id("getPage")
                                .alias("t")
                                .includes(collection=>{
                                        collection.includes(all.bindRecord.columnsArray)
                                })
                                .conditions(collection=>{
                                        collection.includes(all.bindRecord.columnsArray)
                                                  .excludes("id")
                                                  .excludes(["msg",])
                                                  .exp("createTime","timeRange")
                                })
                                .request(request => {
                                        request.path("/bindRecord/page")
                                               .description("get bindRecord page");;
                                })
                                .req(req => {
                                        req.doCreate()
                                           .excludes("id")
                                           .validate("bindStatus","@Enum(\"bindStatus\")")
                                           .name("GetBindRecordPageReq")
                                })
                   
                                .build(),

                        
 ]
}
